From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Outfit7TSB <sunnybunnies09@naver.com>
Date: Tue, 21 Jan 2025 00:48:01 +0900
Subject: [PATCH] remove final from holderable(Paper.utils) +
 ingredient(NMS.item.crafting)

NMS: net.minecraft.* from minecraft vanilla server
Paper: io.papermc.paper
#hardforkisscary

diff --git a/src/main/java/io/papermc/paper/util/Holderable.java b/src/main/java/io/papermc/paper/util/Holderable.java
index 3401d0073f1a98360495c1e73ae2de0146601604..77c4acd13cfcaf0164bf872c157a55195909a00c 100644
--- a/src/main/java/io/papermc/paper/util/Holderable.java
+++ b/src/main/java/io/papermc/paper/util/Holderable.java
@@ -33,7 +33,7 @@ public interface Holderable<M> extends Handleable<M> {
                 }
                 yield registry.get(Key.key(string));
             }
-            case JsonObjectWrapper(final JsonObject element) -> {
+            case JsonObjectWrapper(JsonObject element) -> {
                 if (!(registry instanceof final CraftRegistry<?, ?> craftRegistry) || !craftRegistry.supportsDirectHolders()) {
                     throw new IllegalArgumentException("Cannot deserialize direct holders for " + registry);
                 }
diff --git a/src/main/java/net/minecraft/world/item/crafting/Ingredient.java b/src/main/java/net/minecraft/world/item/crafting/Ingredient.java
index 0b0054b3d5d56ba24e1aee0e3ab56ea5b01a82a8..469b811c7f9d1a40478f1216e80ea0f9e615539e 100644
--- a/src/main/java/net/minecraft/world/item/crafting/Ingredient.java
+++ b/src/main/java/net/minecraft/world/item/crafting/Ingredient.java
@@ -108,9 +108,9 @@ public final class Ingredient implements StackedContents.IngredientInfo<io.paper
     @Override
     public boolean acceptsItem(final io.papermc.paper.inventory.recipe.ItemOrExact holder) {
         return switch (holder) {
-            case io.papermc.paper.inventory.recipe.ItemOrExact.Item(final Holder<Item> item) ->
+            case io.papermc.paper.inventory.recipe.ItemOrExact.Item(Holder<Item> item) ->
                 !this.isExact() && this.values.contains(item);
-            case io.papermc.paper.inventory.recipe.ItemOrExact.Exact(final ItemStack exact) ->
+            case io.papermc.paper.inventory.recipe.ItemOrExact.Exact(ItemStack exact) ->
                 this.isExact() && this.itemStacks.contains(exact);
         };
         // Paper end - Improve exact choice recipe ingredients
